astroquery/heasarc/core.py

```diff
@@ -1,92 +1,246 @@
+
+        Parameters
+        ----------
+        master: bool
+            Select only master tables. Default is False
+        keywords: str or list
+            a str or a list of str of keywords used as search
+            terms for tables. Words with a str separated by a space
+            are AND'ed, while words in a list are OR'ed
+
+        Returns
+        -------
+        `~astropy.table.Table` with columns: name, description
 
-        cache (bool) defaults to True. If set overrides global caching behavior.
-        See :ref:`caching documentation <astroquery_cache>`.
         """
-        request_payload = self._args_to_payload(
-            entry='none',
-            mission='xxx',
-            displaymode='BatchDisplay'
-        )
+        if isinstance(keywords, str):
+            keywords = [keywords]
+            if not all([isinstance(wrd, str) for wrd in keywords]):
+                raise ValueError('non-str found in keywords elements')
+
+        # use 'mast' to include both 'master' and 'mastr'
+        names, desc = [], []
+        for lab, tab in self.tap.tables.items():
+            if 'TAP' in lab or (master and 'mast' not in lab):
+                continue
+            if keywords is not None:
+                matched = any(
+                    [
+                        all([wrd.lower() in f'{lab} {tab.description}'.lower()
+                            for wrd in wrds.split()])
+                        for wrds in keywords
+                    ]
+                )
+                if not matched:
+                    continue
+            names.append(lab)
+            desc.append(tab.description)
+        return Table({'name': names, 'description': desc})
+
+    @deprecated(
+        since='TBD',
+        message=('Heasarc.query_mission_list is deprecated. '
+                 'Use ~Heasarc.tables instead'),
+    )
+    def query_mission_list(self, *, cache=True, get_query_payload=False):
+        """Returns a list of all available mission tables with descriptions.
 
-        if get_query_payload:
-            return request_payload
+        This method is deprecated, and is included only for limited
+        backward compatibility with the old astroquery.Heasarc that uses
+        the Browse interface. Please use ~Heasarc.tables instead.
 
-        # Parse the results specially (it's ascii format, not fits)
-        response = self.query_async(
-            request_payload,
-            url=conf.server,
-            cache=cache
-        )
-        data_str = response.text
-        data_str = data_str.replace('Table xxx does not seem to exist!\n\n\n\nAvailable tables:\n', '')
-        table = Table.read(data_str, format='ascii.fixed_width_two_line',
-                           delimiter='+', header_start=1, position_line=2,
-                           data_start=3, data_end=-1)
-        return table
+        """
+        return self.tables(master=False)
+
+    def columns(self, table_name, full=False):
```

Comment:
Same, make this `list_columns`. 

Also, docstring is not consistent, it says dictionary yet the Returns section lists a Table

Reply:
still stands